### PR TITLE
[1LP][RFR]V2v fixes 510z

### PR DESCRIPTION
**Pull request by kedark3**
Fixing Navigation as cfme 510z has different menu options as well as fixing vm_obj call by replacing it with vm_list

{{pytest: cfme/tests/v2v/test_v2v_migrations.py --use-provider rhv42 --use-provider vsphere67-nested --provider-limit 2 -vvvv --long-running -k 'test_single_datastore_single_vm_migration' }}

### CODE REVIEW

**Comment by izapolsk** [comment]
@kedark3, all PRT tests failed. please give some explanation or check your code

**Comment by kedark3** [comment]
I checked PRT @izapolsk, its being worked on. I'll do WIPTEST.

**Comment by jawatts** [comment]
Moving back to WIPTEST to address failures, Discussing with @kedark3 across the room

**Comment by kedark3** [comment]
@jawatts moving back to RFR, only 1 test failed due to timeout. `.//h2` solved the other issue.

**Comment by jawatts** [comment]
@kedark3 Did you look at the 5.9 results, 0% passing
```Message: Could not find an element Locator(by='xpath', locator='//div[@id="tab_div"]/h3')```

**Comment by kedark3** [comment]
> @kedark3 Did you look at the 5.9 results, 0% passing
> `Message: Could not find an element Locator(by='xpath', locator='//div[@id="tab_div"]/h3')`

This is due to some bug with 59z. It cannot tag the Hosts. Expected to fail.

**Comment by jawatts** [comment]
> > @kedark3 Did you look at the 5.9 results, 0% passing
> > `Message: Could not find an element Locator(by='xpath', locator='//div[@id="tab_div"]/h3')`
> 
> This is due to some bug with 59z. It cannot tag the Hosts. Expected to fail.

Talked offline, tagging bug will be fixed in the next 5.9.6 build. Since the build is expected tomorrow, forgoing adding BZ blocker

File: cfme/v2v/migrations.py [hunk 1]
```diff
@@ -360,21 +381,68 @@ def plan_in_progress(self, plan_name):
...
+            (self.create_infrastructure_mapping.is_displayed or
+                self.infra_mapping_list.is_displayed or self.configure_providers.is_displayed))
+
+
 class AddInfrastructureMappingView(View):
     form = InfraMappingWizard()
+    form_title_text = VersionPicker({
+        Version.lowest(): 'Infrastructure Mapping Wizard',
+        '5.10': 'Create Infrastructure Mapping'
+    })
```
Comment (jawatts) [inline cfme/v2v/migrations.py]: **Question** Why did you move this out, was this going to be used in another method?

File: cfme/common/vm.py [hunk 2]
```diff
@@ -183,7 +183,11 @@ def mac_address(self):
         """Fetches MAC Address of VM"""
         # TODO: We should update this with wrapanapi method when it becomes available.
         view = navigate_to(self, "Details", use_resetter=False)
-        return view.entities.summary('Properties').get_text_of('MAC Address')
+        try:
+            return view.entities.summary('Properties').get_text_of("MAC Address")
+        except NameError:
+            # since some providers have plural 'Addresses'.
+            return view.entities.summary('Properties').get_text_of("MAC Addresses")
```
[the remark this answers is not in the document]
Comment (kedark3) [reply cfme/common/vm.py]: +1

File: cfme/fixtures/v2v.py [hunk 3]
```diff
@@ -138,9 +138,11 @@ def get_vm(request, appliance, second_provider, template, datastore='nfs'):
     vm_obj = collection.instantiate(vm_name,
                                     second_provider,
                                     template_name=template(second_provider)['name'])
-
+    power_on_vm = True
+    if template.__name__ == 'win10_template':
```
[the remark this answers is not in the document]
Comment (kedark3) [reply cfme/fixtures/v2v.py]: +1

File: cfme/v2v/migrations.py (hunk 1 above)
Comment (kedark3) [reply cfme/v2v/migrations.py]: This was failing otherwise. Mike did ask me to debug why, but I didn't get a chance, but current way seems to work fine.

File: cfme/v2v/migrations.py [hunk 4]
```diff
@@ -316,20 +316,41 @@ class MigrationDashboardView(BaseLoggedInPage):
...
+        return ((self.navigation.currently_selected == ['Compute', 'Migration'] or
+            self.navigation.currently_selected == ['Compute', 'Migration', 'Overview']) and
             (len(self.browser.elements(".//div[contains(@class,'spinner')]")) == 0) and
             (len(self.browser.elements('.//div[contains(@class,"card-pf")]')) > 0) and
             len(self.browser.elements(".//div[contains(@class,'pficon-warning-triangle-o')]")) < 1)
 
+    def switch_to(self, section):
+        """Switches to Not Started, In Progress, Complete or Archived Plans section."""
+        sections = {
+            'Not Started Plans': self.not_started_plans,
```
Comment (mshriver) [inline cfme/v2v/migrations.py]: TODO: when refactoring these views to be nested and conditional, let's make these view constants so callers don't have to match a specific string.

Comment (kedark3) [reply cfme/v2v/migrations.py]: I will add that TODO in code. And update the PR.
